tests/assoc_sim.R

```diff
@@ -0,0 +1,143 @@
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+library("optparse")
+
+# First parse args
+options(stringsAsFactors = FALSE)
+if(!interactive()) {
+  option_list <- list(
+    # Flags
+    make_option(c("--candidates", "-c"), default = "candidates.csv",
+                help = "Filename for candidates"),
+    make_option(c("--params", "-p"), default = "params.csv",
+                help = "Filename for parameters"),
+    make_option(c("--reports", "-r"), default = "reports.csv",
+                help = "Filename for reports"),
+    make_option(c("--map", "-m"), default = "map",
+                help = "Filename *prefix* for map(s)"),
+    make_option(c("--num", "-n"), default = 1e05,
+                help = "Number of reports"),
+    make_option(c("--unif", "-u"), default = FALSE,
+                help = "Run simulation with uniform distribution")
+  )
+  opts <- parse_args(OptionParser(option_list = option_list))
+}    
+
+source("../analysis/R/encode.R")
+source("../analysis/R/decode.R")
+source("../analysis/R/simulation.R")
+source("../analysis/R/read_input.R")
+source("../analysis/R/association.R")
+
+# Read candidates from a csv file
+# Inputs: filename. The file is expected to contain two rows of strings
+#         (one for each variable):
+#         "google.com", "apple.com", ...
+#         "ssl", "nossl", ...
+# Returns: a list containing strings
+GetCandidatesFromFile <- function(filename) {
+  contents <- read.csv(filename, header = FALSE)
+  # Expect 2 rows of candidates
+  if(nrow(contents) != 2) {
+    stop(paste("Candidate file", filename, "expected to have
+               two rows of strings."))
+  }
+  list(var1 = as.vector(t(contents[1,])),
+       var2 = as.vector(t(contents[2,])))
+}
+
+# Simulate correlated reports and write into reportsfile
+# Inputs: N = number of reports
+#         candidates = list containing a list of candidate strings
+#         params = list with RAPPOR parameters
+#         mapfile = file to write maps into (with .csv suffixes)
+#         reportsfile = file to write reports into (with .csv suffix)
+#         unif = whether to replace poisson with uniform
+SimulateReports <- function(N, candidates, params, mapfile, reportsfile,
+                            unif) {
+  # Compute true distribution
+  m <- params$m  
+  samples <- list()
+
+  if (unif) {
+    # Draw uniformly from 1 to 10
+    samples[[1]] <- as.integer(runif(N, 1, 10))
+  } else {
+    # Draw from a Poisson random variable
```

Comment:
No, because recall that the single-var distribution analysis of rappor works with aggregated counts. Unfortunately for associations between variables, we require individual reports still. Aggregated counts generated by gen_counts.R will not suffice for now. In the future, me and Ilya want to figure out if we can work with some sort of aggregated data (marginals perhaps). For now, I'll have to rely on individually drawn distributions and not simulated total counts as in gen_counts.R.